### DISCUSSION TITLE
set / change previous menu

### DISCUSSION THREAD

**Discussion by testoverride** [NuSkooler/enigma-bbs#395] [Q&A] [answered]
Is there a way to change the menu that you came from to mainMenu? 

Example: I would like to be able to go to the message  menu from the file menu but when i quit to main on the message menu, it would take you back to the main menu instead of the file menu which is the previous menu.

**Comment by NuSkooler** [comment]
@testoverride Please see https://nuskooler.github.io/enigma-bbs/configuration/menu-hjson.html specifically around the menu flags. I think that should give you what you need. You can peek around some of the default menu layout for example usage.

**Reply by testoverride** [reply]
not sure if those will help. I am trying to insert a quick login menu into the login sequence.  If i use nohistory or popParent it will take me back to the login menu.  I need a way to set the last menu to main menu instead of quicklogon menu when a person goes to another menu from the quicklogon menu otherwise they just go back to the quicklogon menu.

what are the available extraArgs that could be used?


![2022-01-31](https://user-images.githubusercontent.com/3768527/151894720-9708a348-c50a-4399-bd8d-645174b62437.png)

**Comment by NuSkooler** [comment, marked as answer]
> Example: I would like to be able to go to the message menu from the file menu but when i quit to main on the message menu, it would take you back to the main menu instead of the file menu which is the previous menu.

I'm re-reading this again -- If you have say a `mainMenu`, a `messageMenu`, and a `fileMenu` with a link to the `messageMenu` from `fileMenu`, just `action: @systemMethod:prevMenu`. When going from main -> message, the prev is main. When going from file -> message, the prev is file.

For your quick login, create an alternate menu path:
quick -> "message" -> `quickMessageMenu` { quit to main = `mainMenu` }

where your standard `messageMenu` probably has quit to main = `@systemMethod:prevMenu`

**Reply by testoverride** [reply]
that will work.  I had wanted to minimize the additional menus, but it kind of follows how things currently work with the login screens vs main menu ones for oneliners / stats, etc...

my next step is to try to find some time to dive into the ACS setup and get a quicklogin.js for all the login configurable options.

i want the user to be able to select which items in the login process the user will see when doing a normal login and be able to disable everthing except the one that goes to the mainMenu.

Thanks for the replies.

**Comment by NuSkooler** [comment]
> I had wanted to minimize the additional menus some of the reference / shared fragment stuff may help you here.

Would love to see what you come up with for `quicklogin.js`! The AC for querying property values should help you here. Your mod can store user properties (ie: their selection) and use it in later logins.

**Reply by testoverride** [reply]
cool. it will take me while since the log4shell vulnerability is killing us at work.  I should get some free time to try to devote to learning the system and creating the quick login options.  I want a check for everything that is in the current fullLoginSequence so a user can pick and choose what they want t o see when logging into the BBS.

as it gets closer i will open the BBS up to the public, but i got a lot to learn about enigma.
